arrow/arrow.py

```diff
@@ -904,7 +904,13 @@ def _get_iteration_params(cls, end, limit):
     def _get_timestamp_from_input(timestamp):
 
         try:
-            return float(timestamp)
+            timestamp = float(timestamp)
+            if timestamp >= 253402300800000: # nanoseconds
+                return timestamp / 1000000
+            elif timestamp >= 253402300800: # milliseconds
```

Comment:
What does this number represent?

It seems to me that the timestamp `1000` is fully ambiguous with respect to units (does it mean 1000 seconds since 1970 or 1000 nanoseconds?). Since we can't infer the units for all values, it seems futile (invalid) to do it for some values.